javascript/modules/client/src/internal/client-core.ts

```diff
@@ -166,4 +183,58 @@ export abstract class ClientCore implements IClientCore {
       }
     );
   }
+
+  protected async deposit(params: IDeposit): Promise<boolean> {
+    const [
+      daoAddress,
+      amount,
+      tokenAddress,
+      reference,
+    ] = ClientCore.createDepositParameters(params);
+
+    const daoInstance = DAO__factory.connect(daoAddress, this.connectedSigner);
+
+    const override =
+      tokenAddress !== AddressZero
+        ? {}
+        : {
+            value: amount,
+          };
+
+    if (tokenAddress !== AddressZero) {
+      const governanceERC20Instance = GovernanceERC20__factory.connect(
+        tokenAddress,
+        this.connectedSigner
+      );
+
+      const currentAllowance = await governanceERC20Instance.allowance(
+        await this.connectedSigner.getAddress(),
+        daoAddress
+      );
+
+      if (currentAllowance.lt(amount)) {
+        await governanceERC20Instance
+          .increaseAllowance(daoAddress, amount.sub(currentAllowance))
+          .then(tx => tx.wait())
+          .then(cr => {
+            if (
+              !amount.eq(
+                cr.events?.find(e => e?.event === "Approval")?.args?.value
+              )
+            ) {
+              throw new Error("Could not increase allowance");
+            }
+          });
+      }
+    }
+
+    return daoInstance
+      .deposit(tokenAddress, amount, reference, override)
+      .then(tx => tx.wait())
+      .then(cr => {
+        return amount.eq(
```

Comment:
Why returning a boolean?
If something failed, we should return an error, and not `false`